tee-worker/litentry/pallets/identity-management/src/lib.rs

```diff
@@ -243,13 +243,9 @@ pub mod pallet {
 			IDGraphs::<T>::remove(owner, identity);
 		}
 
-		pub fn get_id_graph(who: &T::AccountId) -> IDGraph<T> {
-			IDGraphs::<T>::iter_prefix(who).collect::<_>()
-		}
-
 		// get the most recent `max_len` elements in IDGraph
-		pub fn get_id_graph_with_max_len(who: &T::AccountId, max_len: usize) -> IDGraph<T> {
-			let mut id_graph = Self::get_id_graph(who);
+		pub fn get_id_graph(who: &T::AccountId, max_len: usize) -> IDGraph<T> {
+			let mut id_graph = IDGraphs::<T>::iter_prefix(who).collect::<IDGraph<T>>();
 			id_graph.sort_by(|a, b| Ord::cmp(&b.1.link_block, &a.1.link_block));
```

Comment:
I think this can still lead to non deterministic ordering but with much lower probability (several identities created within same block).

Reply:
Strictly speaking that's true - but that should only happen very rarely without batching